templates/cli/lib/commands/deploy.js.twig

```diff
@@ -450,12 +453,54 @@ const deployFunction = async ({ functionId, all, yes } = {}) => {
                     updaterRow.fail({ errorMessage: e.message ?? 'General error occurs please try again' })
             }
         }
+
+        if (deploymentCreated) {
+            try {
+                const deploymentId = response['$id'];
+                updaterRow.update({ status: 'Deploying', end: 'Checking deployment status...' })
+                let pool_checks = 0;
+
+                while (true) {
+                    if (pool_checks >= POOL_MAX_DEBOUNCE) {
+                        updaterRow.update({ end: 'Deployment takes too long; check the console' })
+                        break;
+                    }
+
+                    response = await functionsGetDeployment({
+                        functionId: func['$id'],
+                        deploymentId: deploymentId,
+                        parseOutput: false
+                    });
+
+
+                    const status = response['status'];
+                    if (status === 'ready') {
+                        updaterRow.update({ status: 'Deployed' });
+                        successfullyDeployed++;
+
+                        break;
+                    } else if (status === 'failed') {
+                        updaterRow.fail({ errorMessage: 'failed to deploy' });
+
+                        break;
+                    } else {
+                        updaterRow.update({ status: 'Deploying', end: `current status: ${status}` })
+                    }
+
+                    pool_checks++;
+                    await new Promise(resolve => setTimeout(resolve, POOL_DEBOUNCE));
+                }
+            } catch (e) {
+                updaterRow.fail({ errorMessage: e.message ?? 'General error occurs please try again' })
+            }
+        }
+
         updaterRow.stopSpinner();
     }));
 
     Updater.stop();
 
-    success(`Pushed ${successfullyDeployed} functions`);
+    success(`Pushed ${successfullyPushed} functions with ${successfullyDeployed} successfully deployed`);
```

Comment:
```suggestion
    success(`Successfully pushed ${successfullyPushed} functions.`);
```
I dont think the `successfullyDeployed` variable is required here?

Reply:
The `successfullyDeployed` inform how many functions didn't fail in the Appwrite instance, could be changed to 
```js
success(`Successfully pushed & deployed ${successfullyDeployed} functions.`);
```

Reply:
Or, we can use the refactor success message but only for deployed (pushed + deployed) functions
```js
success(`Successfully pushed ${successfullyDeployed} functions.`);
```